forte/data/data_pack.py

```diff
@@ -610,6 +639,36 @@ def set_audio(
         ap.set_cache(audio)
         ap.sample_rate = sample_rate
 
+    def set_image(
+        self,
+        image,
+        image_payload_index: int = 0,
+    ):
+        r"""Set the image payload of the :class:`~forte.data.data_pack.DataPack`
+        object.
+
+        Args:
+            image: A numpy array storing the image.
+            image_payload_index: the zero-based index of the ImagePayload
+                in this DataPack's ImagePayload entries. Defaults to 0.
+        """
+        # temporary solution for backward compatibility
+        # past API use this method to add a single image in the datapack
+        if (
+            self._data_store.num_entries("ft.onto.base_ontology.ImagePayload")
+            == 0
+            and image_payload_index == 0
+        ):
```

Comment:
instead of only providing the comment, we should provide a log saying for backward compatibility we are setting the first image when there is nothing and image_payload_index=0, however, we should encourage users to use `add_image` function.

Let's add these notes to both `image` and `audio`, and avoid using `set_image` and `set_audio` in this way (only use them when the index actually exists) in our own tutorials/examples.

For `set_text`, since it is widely used in the repo, let's not add the logging to it for now.

Reply:
Added logging info and `add_image` function.

Reply:
Maybe we should also add these notes to the `set_audio` function. And do we need an `add_audio` function as well?

Reply:
https://github.com/asyml/forte/issues/895